File: Python/game_scripts/display.py
# For checking width of unicode characters
# import wcwidth
def new_line(num):
    for _ in range(num):
        print("\n")


def display_array(
    castle,
    game_sequence,
    player=None,
    monster_hp=None,
    hp=None,
    mana=None,
    money=None,
    floor=None,
):
    new_line(50)
    if game_sequence in ["non-battle", "battle"]:
        hp = player.hp
        mana = player.mana
        money = player.money
        floor = player.floor
    new_line(1)
    print(f"Floor {floor + 1} of {len(castle)}")
    new_line(1)
    # print('\n'.join(['  '.join(['{:^{}}'.format(cell,wcwidth.wcwidth(u'\U0001f6aa')) if cell == '' else cell for cell in row]) for row in castle[floor]]))
    print(
        "\n".join(
            [
                "  ".join(
                    ["{:^{}}".format(cell, 2) if cell == "" else cell for cell in row]
                )
                for row in castle[floor]
            ]
        )
    )
    new_line(1)
    if not game_sequence == "battle":
        print(f"HP: {hp}% | Mana: {mana}% | \U0001f4b0: {money}")
    else:
        print(
            f"HP: {hp}% | Mana: {mana}% | \U0001f4b0: {money} | Monster HP: {monster_hp}"
        )
    new_line(1)


if __name__ == "__main__":
    print("You must run 'python3 start_game.py' to play Castle Descent.")

File: dev/RStudio/src/display.R
# TODO: Refactor
display_array <- function(castle_data,player,floor,total_floors,mana,hp,money,game_sequence,monster_hp){
  new_line(50)
  if(!game_sequence == "next level"){
    floor <- player$floor 
    total_floors <- player$total_floors
    hp <- player$hp
    mana <- player$mana
    money <- player$money
  }
  cat(sprintf("Floor %s of %s",floor,total_floors))
  new_line(2)
  print(castle_data$castle[,,floor],quote = F)
  new_line(1)
  if(game_sequence %in% c("non-battle", "next level")){
    #Display Health information and money
    cat(sprintf("HP: %s %s Mana: %s %s %s: %s",paste0(hp,"%"),"|",paste0(mana,"%"), "|",money_unicode <- "\U1F4B0", money))
  }else{
    cat(sprintf("HP: %s %s Mana: %s %s Monster HP: %s",paste0(hp,"%"),"|",paste0(mana,"%")
                ,"|",monster_hp))
  }

  new_line(2)

}

new_line <- function(num){
  cat(rep("\n", num))
}